d2ir/compile.go

```diff
@@ -127,13 +129,74 @@ func (c *compiler) compileField(dst *Map, kp *d2ast.KeyPath, refctx *RefContext)
 		}
 		c.compileMap(f.Map(), refctx.Key.Value.Map)
 	} else if refctx.Key.Value.ScalarBox().Unbox() != nil {
+		// If these are boards, we transform into absolute paths
+		if f.Name == "link" {
+			c.compileLink(refctx)
+		}
 		f.Primary_ = &Scalar{
 			parent: f,
 			Value:  refctx.Key.Value.ScalarBox().Unbox(),
 		}
 	}
 }
 
+func (c *compiler) compileLink(refctx *RefContext) {
+	val := refctx.Key.Value.ScalarBox().Unbox().ScalarString()
+	link, err := d2parser.ParseKey(val)
+	if err != nil {
+		return
+	}
+
+	scopeID, _ := d2parser.ParseKey(refctx.ScopeMap.AbsID())
+	scopeIDA := scopeID.IDA()
+
+	if len(scopeIDA) == 0 {
+		return
+	}
+
+	linkIDA := link.IDA()
+	if len(linkIDA) == 0 {
+		return
+	}
+
+	// If it doesn't start with one of these reserved words, the link may be a URL or local path or something
+	if linkIDA[0] != "layers" && linkIDA[0] != "scenarios" && linkIDA[0] != "steps" && linkIDA[0] != "_" {
+		return
+	}
+
+	// Chop off the non-board portion of the scope, like if this is being defined on a nested object (e.g. `x.y.z`)
+	for i := len(scopeIDA) - 1; i > 0; i-- {
```

Comment:
```suggestion
	for i := len(scopeIDA) - 1; i > 1; i-- {
```

Reply:
`link-board-key-nested` test fails with this change:

```
x: {
  y.link: layers.x
}
layers: {
  x: {
    yo
  }
}
```

the scope of `x.y.link` should be `root`. this loop needs to go all the way to 0th, and see `root`, and chop everything after

Reply:
Oh right, sorry I misunderstood this loop, I thought it was >= 0, thought it'd panic at 0 but nah it's fine.